container.go

```diff
@@ -96,7 +96,7 @@ type Container interface {
 	//
 	// errors:
 	// Systemerror - System error.
-	Set() error
+	Set(config configs.Config) error
```

Comment:
I think Set could have more functionality in future to cover more than cgroups, so taking in the entire config makes sense to me.